### PR TITLE
add upgrade e2e test

### PR DESCRIPTION
**Pull request by xiaojingchen**
this pr add e2e test case to check upgrade feature
PTAL

### CODE REVIEW

File: tests/e2e/upgrade.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+}
+
+func upgrade() (bool, error) {
+	tc, err := cli.PingcapV1alpha1().TidbClusters(ns).Get(clusterName, metav1.GetOptions{})
+	if err != nil {
+		logf("failed to upgrade tidbcluster, error: %v", err)
+		return false, err
+	}
+
+	tc.Spec.PD.Image = strings.Replace(tc.Spec.PD.Image, getImageTag(tc.Spec.PD.Image), "v2.0.0", -1)
```
Comment (tennix) [inline tests/e2e/upgrade.go]: Do we have TiDB `v2.0.0` docker images in our CI? It's better to define version as a constant.

Comment (xiaojingchen) [reply tests/e2e/upgrade.go]: I have pushed v2.0.0 to registry manually.

File: tests/e2e/upgrade.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	. "github.com/onsi/gomega"
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap.com/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/controller"
+	apps "k8s.io/api/apps/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	upgradeVersion = "v2.0.0"
```
Comment (weekface) [inline tests/e2e/upgrade.go]: use v2.0.5 instead, downgrade may be unsafe

Comment (xiaojingchen) [reply tests/e2e/upgrade.go]: the v2.0.5 version is not exist

**Pull request merged**